### PR TITLE
Select trace color

### PR DESCRIPTION
**Pull request by fanmingyu212**
Addresses #9. Added a submenu in the context menu of the trace list widget to select trace colors.

### CODE REVIEW

File: TraceListWidget.py [hunk 1]
```diff
@@ -44,11 +44,11 @@ def popupMenu(self, pos):
...
-        action = menu.exec_(self.mapToGlobal(pos))
-        if action == dataaddAction:
-            dvlist = DataVaultList(self.parent.name)
-            self.windows.append(dvlist)
-            dvlist.show()
+            action = menu.exec_(self.mapToGlobal(pos))
+            if action == dataaddAction:
+                dvlist = DataVaultList(self.parent.name)
+                self.windows.append(dvlist)
+                dvlist.show()
```
Comment (aransfor) [inline TraceListWidget.py:52]: what is going on here? can you explain why this logic need to be inside the if (item==None): statement?

File: TraceListWidget.py [hunk 2]
```diff
@@ -80,5 +87,49 @@ def popupMenu(self, pos):
...
+                    self.parent.artists[ident].artist.setData(pen = new_color, symbolBrush = new_color)
+                else:
+                    self.parent.artists[ident].artist.setData(pen = new_color)
+
+            if colorselctionAction == whiteAction:
+                new_color = "w"
+                if self.parent.show_points:
+                    self.parent.artists[ident].artist.setData(pen = new_color, symbolBrush = new_color)
+                else:
+                    self.parent.artists[ident].artist.setData(pen = new_color)
```
Comment (aransfor) [inline TraceListWidget.py]: For this code I would suggest having a single if statement and a dictionary example:

```python

coloractiondict = {magentaAction:'m', whitAction:'w'}

if colorselectionAction in coloractiondict.keys():
       new_color = coloractiondict[colorselectionAction]
....
...
...

```

**Comment by aransfor** [comment]
@fanmingyu212 also you have not pushed this branch to the cloud so I cant check it out it seems.

File: TraceListWidget.py (hunk 1 above)
Comment (fanmingyu212) [reply TraceListWidget.py:52]: If this logic is not inside of if (item == None), dataaddAction is not defined when python executes line 48. I think I didn't change any logic here. The original code uses a mixture of tab and spaces so the white space offset in the old code may not show the actual logic of the code.

Comment (fanmingyu212) [reply TraceListWidget.py]: If this logic is not inside of if (item == None), dataaddAction is not defined when python executes line 48. I think I didn't change any logic here. The original code uses a mixture of tab and spaces so the white space offset in the old code may not show the actual logic of the code.

File: TraceListWidget.py (hunk 2 above)
Comment (fanmingyu212) [reply TraceListWidget.py]: Okay. That will simplify the code. Will change it as requested.

**Comment by fanmingyu212** [comment]
@aransfor You can access the branch at https://github.com/fanmingyu212/RealSimpleGrapher/tree/mf/select_trace_color. It is in a folked repo because I don't have the write access to this repo.

**Comment by aransfor** [comment]
@fanmingyu212 it appears I dont have access to your forked branch, could you please add me so I can check the code out

**Comment by fanmingyu212** [comment]
@aransfor Made changes as requested. I invited you to collaborate on the folked repository. Please let me know if you still cannot checkout branch.

**Comment by aransfor** [comment]
LGTM

**Pull request merged**